tip.lic

```diff
@@ -0,0 +1,85 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#tip
+=end
+
+custom_require.call %w[common common-money]
+
+class Tip
+
+  # include DRC for the bput function
+  # include DRCM for various money handling functions
+  
+  def initialize
+    arg_definitions = [
+      # Two required arguments - recipient and amount; One optional argument, currency
+      # Mirrors DR's mechanics of "tip <person> 10000 kron", and also allows "tip <person> 10000" and assumes coppers
+      [
+        { name: 'recipient', regex: /^[A-z\.\s\-']+$/i, variable: true, description: 'The recipent of your tip. Abbreviations ok.' },
+        { name: 'amount', regex: /^\d+$/i, variable: true, description: 'Numeric amount to tip, no commas (e.g. the number "100" in 100 kronars)' },
+        { name: 'currency', regex: Regexp.union($CURRENCY_REGEX_MAP.values), optional: true, variable: true, description: 'Optional currency. Abbreviations ok: kron, lirums, dokor' }         
+      ],
+      # Three required arguments - recipient, amount, denomination - plus a fourth optional argument - currency
+      # Allows mechanic of "tip person 10 plat" and chooses your current max on hand
+      # or "tip person 10 plat kron"
+      [
+        { name: 'recipient', regex: /^[A-z\.\s\-']+$/i, variable: true, description: 'The recipent of your tip. Abbreviations ok.' },
+        { name: 'amount', regex: /^\d+$/i, variable: true, description: 'Numeric amount to tip, no commas (e.g. the number "100" in 100 kronars)' },
+        { name: 'denomination', regex: Regexp.union($DENOMINATION_REGEX_MAP.values), variable: true, description: 'The denomination. Abbrevations ok. (e.g. plat, gold, silv)' },
+        { name: 'currency', regex: Regexp.union($CURRENCY_REGEX_MAP.values), optional: true, variable: true, description: 'Optional currency. Abbreviations ok: kron, lirums, dokor' }
+      ]
+    ]
+    
+    args = parse_args(arg_definitions)  
+    gratuity(args.recipient, args.amount, args.denomination, args.currency)
+
+  end
+
+  @tip_success_patterns = [
+    /^You offer/i
+  ]
+
+  @tip_failure_patterns = [
+    /^I don't know who/i,
+    /you really should keep every bronze/i,
+    /^You already have a tip offer outstanding/i,
+    /^But you don't have that much/i,
+    /^Hmmm, tipping yourself/i    
+  ]
+
+  def gratuity(recipient, amount, denomination, currency)    
+    # Grab the total amount in copper given a numerical amount
+    # and a denomination (e.g. 100 bronze). Utilize the
+    # helper method in common-money.
+    total = DRCM.convert_to_copper(amount, denomination)
```

Comment:
💡 To now avoid the [warning](https://github.com/rpherbig/dr-scripts/pull/5271/files#diff-7dae83dd2163ccf532277ddc13167af83188df20c3bdccd306234d70ec92eb34R53) from common-money's `convert_to_copper` about unknown denomination, `Tip` can provide a reasonable default of `coppers`.

```suggestion
    denomination ||= 'copper'
    total = DRCM.convert_to_copper(amount, denomination)
```

Reply:
Thanks for this. Neat reading on double-pipe equals, and truthiness, falseyness.